spec/support/devise.rb

```diff
@@ -2,4 +2,5 @@
 
 RSpec.configure do |config|
   config.include Warden::Test::Helpers
```

Comment:
you probably don't need both :). Read please docs and search which one is truly needed

Reply:
Hey, how about that?

https://gist.github.com/boddhisattva/f14d2e860dcd2d45f805eb698b99605f

Specify the below helpers in the RSpec.configure do |config| block of your rails_helper.rb to make sure that the sign_in method is used to login the user into your app before testing the specific feature of your app which requires a signed in user

```
config.include Devise::Test::IntegrationHelpers, type: :system
config.include Warden::Test::Helpers
```